tests/signal_manager_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+. ./src/signal_manager.sh --source-only
+. ./tests/utils --source-only
+
+function suite() {
+  suite_addTest "default_handler_Test"
+  suite_addTest "non_signal_Test"
+  suite_addTest "new_signal_Test"
+  suite_addTest "signal_reset_Test"
+  suite_addTest "add_two_signals_Test"
+}
+
+function default_handler_Test() {
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+  local output
+
+  cd "$SHUNIT_TMPDIR"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
+    wait $!
+  } > out.tmp &
+
+  $(echo :)
+  kill -s SIGTERM $!
+  assertEquals "$expected" "$(cat out.tmp)"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
+    wait $!
+  } > out.tmp &
+
+  $(echo :)
+  kill -s SIGINT $!
+  assertEquals "$expected" "$(cat out.tmp)"
+
+  cd "$original_dir"
+}
+
+
+function non_signal_Test() {
+  signal_manager 'echo hello!' SIGNONEXISTENT
+  local ret=$?
+  assertFalse "Should have received an error." "$ret"
+}
+
+function new_signal_Test() {
+  local test_phrase="This is a phrase."
+  local original_dir="$PWD"
+  cd "$SHUNIT_TMPDIR"
+
+  set -m
```

Comment:
`set` is a super powerful command, and it is hard to know its options. Add a simple comment explaining what `-m` mean and explain why you need it if it is not clear.